src/main/java/org/passay/LengthComplexityRule.java

```diff
@@ -322,7 +316,7 @@ public boolean includes(final int i)
      */
     public boolean includes(final Interval i)
     {
-      return includes(i.getLowerBoundClosed()) && includes(i.getUpperBoundClosed());
+      return includes(i.lower) && includes(i.upper);
```

Comment:
I think the previous impl makes more sense. Or `boundsPattern.hashCode()` in the new impl.